File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/LoadAverageConverter.java
package com.platdmit.simplecloudmanager.domain.converters;

import com.platdmit.simplecloudmanager.data.api.models.ApiLoadAverage;
import com.platdmit.simplecloudmanager.data.database.entity.DbLoadAverage;
import com.platdmit.simplecloudmanager.domain.models.LoadAverage;

import java.util.List;

public interface LoadAverageConverter {
    List<DbLoadAverage> fromApiToDb(ApiLoadAverage apiLoadAverage, long serverId);
    List<LoadAverage> fromApiToDomain(ApiLoadAverage apiLoadAverage);
    List<LoadAverage> fromDbToDomain(List<DbLoadAverage> dbLoadAverages);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/LoadAverageConverter.kt
package com.platdmit.simplecloudmanager.domain.converters

import com.platdmit.data.api.models.ApiLoadAverage
import com.platdmit.data.database.entity.DbLoadAverage
import com.platdmit.simplecloudmanager.domain.models.LoadAverage

interface LoadAverageConverter {
    fun fromApiToDb(apiLoadAverage: ApiLoadAverage, serverId: Long): List<DbLoadAverage>
    fun fromApiToDomain(apiLoadAverage: ApiLoadAverage): List<LoadAverage>
    fun fromDbToDomain(dbLoadAverages: List<DbLoadAverage>): List<LoadAverage>
}